src/main/webapp/app/exercises/shared/result/result-detail.component.ts

```diff
@@ -183,6 +184,29 @@ export class ResultDetailComponent implements OnInit {
         }
     };
 
+    /**
+     * computes the feedback preview for feedback texts with multiple lines or feedback that is longer than 300 characters
+     * @param text the feedback.detail Text
+     * @return the preview text (one line of text with at most 300 characters)
+     */
+    private static computeFeedbackPreviewText(text: string | undefined): string | undefined {
+        if (text) {
+            if (text.includes('\n')) {
+                // if there are multiple lines, only use the first one
+                const firstLine = text.substr(0, text.indexOf('\n'));
+                if (firstLine.length > 300) {
+                    return firstLine.substr(0, 300);
+                } else {
+                    return firstLine;
+                }
+            } else if (text.length > 300) {
+                return text.substr(0, 300);
```

Comment:
I think that using a variable would be better here.
```suggestion
                const displayCharacterLimit = 300;
                if (firstLine.length > displayCharacterLimit) {
                    return firstLine.substr(0, displayCharacterLimit);
                } else {
                    return firstLine;
                }
            } else if (text.length > displayCharacterLimit) {
                return text.substr(0, displayCharacterLimit);
```

Reply:
Could be a class constant. It would then be easier to find.